### PR TITLE
add nix development shell and lock node version

### PR DESCRIPTION
**Pull request by nyxkrage**
This adds a .nvmrc that pins the node version to 20.5.1, the current latest version, and the version that will be promoted to LTS in the near future
This adds a shell.nix and flake.nix file that specifies the depedencies, node and yarn, and uses the version in the .nvmrc and allows users of the nix package manager to simple run `nix develop` or `nix-shell` to be ready to develop on the project.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  description = "booker-flake";
+
+  inputs = { 
+    nixpkgs.url = "nixpkgs/nixos-unstable";
```
Comment (twny) [inline flake.nix:5]: why did we pick nix `unstable`

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { 
+          inherit system; 
+        };
+
+        buildNodeJs = pkgs.callPackage "${nixpkgs}/pkgs/development/web/nodejs/nodejs.nix" {
+          python = pkgs.python3;
```
Comment (twny) [inline flake.nix:17]: why do we need python to build node?

File: shell.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ pkgs ? import <nixpkgs> {}
+, buildNodeJs ? pkgs.callPackage <nixpkgs/pkgs/development/web/nodejs/nodejs.nix> { python = pkgs.python3; }
```
Comment (twny) [inline shell.nix:2]: can this on it's own line so it's easier to read? i don't know about any nix linters but when they are all on one line i get scarwed

File: shell.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}: let
+  nodejsVersion = pkgs.lib.fileContents ./.nvmrc;
```
Comment (twny) [inline shell.nix:4]: if i don't use nvm, and i use nodenv, will this still work?

**Review by twny — APPROVED** [review]
this looks incredible i really want to use nix in booker! but i think i may need help. so if it breaks can we pair program on this?

File: flake.nix (hunk 1 above)
Comment (nyxkrage) [reply flake.nix:5]: this is to make sure we have the newest version yarn that is in nixpkgs

File: flake.nix (hunk 2 above)
Comment (nyxkrage) [reply flake.nix:17]: because the creators of nodejs decided that python should be a dependency

File: shell.nix (hunk 4 above)
Comment (nyxkrage) [reply shell.nix:4]: nodevn will by default not respect the .nvmrc, but will with a separate plugin https://github.com/ouchxp/nodenv-nvmrc, though both nvm and fnm support it

File: shell.nix (hunk 3 above)
Comment (nyxkrage) [reply shell.nix:2]: I dont know what you are saying here, this is the standard formatting of nix, putting each part of the attrset arguement to the function on a new line with leadings comma

**Pull request merged**